src/pygeoprocessing/geoprocessing.py

```diff
@@ -3788,3 +3584,352 @@ def raster_to_numpy_array(raster_path, band_id=1):
+    target_raster_x_size, target_raster_y_size = target_raster_info[
+        'raster_size']
+    for (raster_path, raster_band_id), resample_method in zip(
+            base_raster_path_band_list, resample_method_list):
+        LOGGER.info(
+            f'stitching {(raster_path, raster_band_id)} into '
+            f'{target_stitch_raster_path_band}')
+        raster_info = get_raster_info(raster_path)
+
+        projected_raster_bounding_box = transform_bounding_box(
+            raster_info['bounding_box'],
+            raster_info['projection_wkt'],
+            target_raster_info['projection_wkt'])
+
+        try:
+            # merge the bounding boxes only to see if they don't intersect
+            _ = merge_bounding_box_list(
+                [projected_raster_bounding_box,
+                 target_raster_info['bounding_box']], 'intersection')
+        except ValueError:
+            LOGGER.warning(
+                f'the raster at "{raster_path}"" does not intersect the '
+                f'stitch raster at "{target_stitch_raster_path_band[0]}", '
+                'skipping...')
+            continue
+
+        # use this to determine if we need to warp and delete if we did at
+        # the end
+        if (raster_info['projection_wkt'] ==
+            target_raster_info['projection_wkt'] and
+            raster_info['pixel_size'] ==
+                target_raster_info['pixel_size']):
+            warped_raster = False
+            base_stitch_raster_path = raster_path
+        else:
+            workspace_dir = tempfile.mkdtemp(
+                dir=os.path.dirname(target_stitch_raster_path_band[0]),
+                prefix='stitch_rasters_workspace')
+            base_stitch_raster_path = os.path.join(
+                workspace_dir, os.path.basename(raster_path))
+            warp_raster(
+                raster_path, target_raster_info['pixel_size'],
+                base_stitch_raster_path, resample_method,
+                target_projection_wkt=target_raster_info['projection_wkt'],
+                working_dir=workspace_dir,
+                osr_axis_mapping_strategy=osr_axis_mapping_strategy)
+            warped_raster = True
+
+        if warped_raster and area_weight_m2_to_wgs84:
+            # determine base area per pixel currently and area per pixel
+            # once it is projected to wgs84 pixel sizes
+            base_pixel_area_m2 = abs(numpy.prod(raster_info['pixel_size']))
+            base_stitch_raster_info = get_raster_info(
+                base_stitch_raster_path)
+            lat_min, lat_max = [
+                base_stitch_raster_info['bounding_box'][_]
+                for _ in [1, 3]]
+            n_rows = base_stitch_raster_info['raster_size'][1]
+            # this column is a longitude invariant latitude variant pixel
+            # area for scaling area dependent values
+            m2_area_per_lat = _create_latitude_m2_area_column(
+                lat_min, lat_max, n_rows)
```

Comment:
Neat.  I was about to comment about the possibility of a `MemoryError` here (like if we had super high-res outputs), but I suppose we can handle that case when it happens.  Was that your thinking as well?

Reply:
Memory error is not a problem-- that's the beauty of this. This function returns a 1 dimensional column of areas because the areas are not variant with longitude. This is the motivation for why I changed `raster_calculator` a couple years ago to allows brodcastable arrays rather than just arrays that are the correct dimensions. `raster_calculator` will slice and broadcast as needed to fit whatever raster block its working on.

Reply:
> This is the motivation for why I changed raster_calculator a couple years ago to allows brodcastable arrays rather than just arrays that are the correct dimensions. raster_calculator will slice and broadcast as needed to fit whatever raster block its working on.

Right, I totally get that! so we know that `raster_calculator` won't `MemoryError`, which is great.  The `MemoryError` I was referring to was about:

> This function returns a 1 dimensional column of areas because the areas are not variant with longitude.

Imagine if we had a base raster with millions of rows.  `__create_latitude_m2_area_column` will happily create a 1 x n column of pixel areas.  Are we at risk of `MemoryError` on that?  Feels like a bit of an edge case (and even less likely now that InVEST binaries use 64-bit builds), but what do you think?

Reply:
I really think that issue is okay. If you have millions of rows, that's still only a few megabytes (estimating 4 bytes * millions numbers of bytes). I'd start to worry about a billion rows but at that point you're talking about a resolution on Earth of about .02m/pixel (estimating 180/1000000000*111km/degree at equator).  But even there you'd still only use 8gb to store the row but you'd be dealing with a 1billion X 2billion element raster on disk that would be on the order of petabytes of hardrive space... 

Am I doing my math right? I'm trying to make the case that this an order of magnitude problem that cannot be exceeded even with crazy numbers, and by the time we got to the technology level that makes that crazy the norm, we'd still only have 8gb array in memory. Which I'm sure would be fine with the many terabytes of local ram we'd have.

Does that sound right?

Reply:
Yep, yep, that sounds right to me!  And if we're working with petabyte-size rasters, we'll run into other more practical problems (like that we'll be pushing the upper limit of number of pixels supported by GeoTiff) before this ends up being a memory issue.